libraries/Netdump/src/NetdumpPacket.h

```diff
@@ -0,0 +1,325 @@
+
+    uint32_t getTcpSeq() const
+    {
+        return ntoh32(ETH_HDR_LEN + getIpHdrLen() + 4);
+    }
+    uint32_t getTcpAck() const
+    {
+        return ntoh32(ETH_HDR_LEN + getIpHdrLen() + 8);
+    }
+    uint16_t getTcpFlags() const
+    {
+        return ntoh16(ETH_HDR_LEN + getIpHdrLen() + 12);
+    }
+    uint16_t getTcpWindow() const
+    {
+        return ntoh16(ETH_HDR_LEN + getIpHdrLen() + 14);
+    }
+    uint8_t  getTcpHdrLen() const
+    {
+        return (data[ETH_HDR_LEN + getIpHdrLen() + 12] >> 4) * 4;
+    };//Header len is in multiple of 4 bytes
+    uint16_t getTcpLen() const
+    {
+        return getIpTotLen() - getIpHdrLen() - getTcpHdrLen() ;
+    };
+
+    uint8_t  getIcmpType() const
+    {
+        return data[ETH_HDR_LEN + getIpHdrLen() + 0];
+    }
+    uint8_t  getIgmpType() const
+    {
+        return data[ETH_HDR_LEN + getIpHdrLen() + 0];
+    }
+
+    uint8_t getARPType() const
+    {
+        return data[ETH_HDR_LEN + 7];
+    }
+    bool    is_ARP_who() const
+    {
+        return getARPType() == 1;
+    }
+    bool    is_ARP_is() const
+    {
+        return getARPType() == 2;
+    }
+
+    uint8_t getUdpHdrLen() const
+    {
+        return 8;
+    };
+    uint16_t getUdpLen() const
+    {
+        return ntoh16(ETH_HDR_LEN + getIpHdrLen() + 4);
+    };
+    bool isARP() const
+    {
+        return (ethType() == 0x0806);
+    };
+    bool isIPv4() const
+    {
+        return (ethType() == 0x0800);
+    };
+    bool isIPv6() const
+    {
+        return (ethType() == 0x86dd);
+    };
+    bool isIP() const
+    {
+        return (isIPv4() || isIPv6());
+    };
+    bool isICMP() const
+    {
+        return ((ipType() == 1) || ipType() == 58);
+    };
+    bool isIGMP() const
+    {
+        return ipType() == 2;
+    };
+    bool isTCP() const
+    {
+        return ipType() == 6;
+    };
+    bool isUDP() const
+    {
+        return ipType() == 17;
+    };
+    bool isMDNS() const
+    {
+        return hasPort(5353);
+    };
+    bool isDNS() const
+    {
+        return hasPort(53);
+    };
+    bool isSSDP() const
+    {
+        return hasPort(1900);
+    };
+    bool isDHCP() const
+    {
+        return (hasPort(546) || hasPort(547) || hasPort(67) || hasPort(68));
+    };
+    bool isWSDD() const
+    {
+        return (hasPort(3702));
+    };
+    bool isHTTP() const
+    {
+        return (hasPort(80));
+    };
+    bool isOTA() const
+    {
+    	return (hasPort(8266));
+    }
+    bool isNETBIOS() const
+    {
+    	return (hasPort(137) || hasPort(138) || hasPort(139));
+    }
+    bool isSMB() const
+    {
+    	return (hasPort(445));
+    }
+    NetdumpIP getIP(uint16_t idx) const
+    {
+        return NetdumpIP(data[idx],
+                         data[idx + 1],
+                         data[idx + 2],
+                         data[idx + 3]);
+    };
+
+    NetdumpIP getIP6(uint16_t idx) const
+    {
+        return NetdumpIP((const uint8_t*)&data[idx], false);
+    };
```

Comment:
Do not understand this remark.
Can you clarify ?

Reply:
I think @devyte says that the IPv6 code in IPAddress and the one here should be factorized.
(Maybe?) this can be done (afterwards?) with a template class since
- we'd like to monitor IPv6 traffic even if IPv6 is not enabled (?)
- IPAddress IPv6's code is based on lwIP (unreachable code when IPv6 is disabled at compile time)

Reply:
NetdumpIP needed for IPv6 address support when using only IPv4